R/math_test_func.R

```diff
@@ -0,0 +1,118 @@
+
+# Modified measure class, now responsible for evaluating the mathematical
+# test function
+MeasureMathTestFun = R6Class("MeasureMathTestfun",
+  inherit = MeasureRegr,
+  public = list(
+    test_fun = NULL,
+
+    initialize = function(test_fun) {
+      self$test_fun = test_fun
+
+      # init superclass
+      super$initialize(
+        id = "func_value",
+        range = c(-Inf, Inf),
+        minimize = TRUE
```

Comment:
Do all test functions in fact need to be minimized?

Reply:
yes, see `getGlobalOpt`: https://cran.r-project.org/web/packages/globalOptTests/globalOptTests.pdf